### PR TITLE
Add CVE conversion for Debian

### PR DESCRIPTION
**Pull request by hogo6002**
Combining the full Debian Security Tracker (excludes `not yet assigned` and `end-of-life`) to our existing CVE entries to address potential data gaps in OSV.dev. 
Adding an `ecosystem_specific` field to indicate the urgency, such as `unimportant`.

### CODE REVIEW

File: vulnfeeds/vulns/vulns.go [hunk 1]
```diff
@@ -163,10 +164,11 @@ func (affected *Affected) AttachExtractedVersionInfo(version cves.VersionInfo) {
...
 type PackageInfo struct {
-	PkgName     string           `json:"pkg_name,omitempty" yaml:"pkg_name,omitempty"`
-	Ecosystem   string           `json:"ecosystem,omitempty" yaml:"ecosystem,omitempty"`
-	PURL        string           `json:"purl,omitempty" yaml:"purl,omitempty"`
-	VersionInfo cves.VersionInfo `json:"fixed_version,omitempty" yaml:"fixed_version,omitempty"`
+	PkgName           string            `json:"pkg_name,omitempty" yaml:"pkg_name,omitempty"`
+	Ecosystem         string            `json:"ecosystem,omitempty" yaml:"ecosystem,omitempty"`
+	PURL              string            `json:"purl,omitempty" yaml:"purl,omitempty"`
+	VersionInfo       cves.VersionInfo  `json:"fixed_version,omitempty" yaml:"fixed_version,omitempty"`
+	EcosystemSpecific map[string]string `json:"ecosystem_specific,omitempty" yaml:"ecosystem_specific,omitempty"`
```
Comment (another-rex) [inline vulnfeeds/vulns/vulns.go:171]: Is there a reason this ecosystem specific field is duplicated here? (already added one in Affected, and the schema only has one in affected, not package)

File: vulnfeeds/cmd/debian/main.go [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+			var pkgInfos []vulns.PackageInfo
+			if value, ok := osvPkgInfos[cveId]; ok {
+				pkgInfos = value
+			}
+
+			for releaseName, release := range cve.Releases {
+				if release.Urgency == "not yet assigned" {
+					continue
+				}
+				debianVersion := getDebianVersion(releaseName)
```
Comment (another-rex) [inline vulnfeeds/cmd/debian/main.go:66]: Can you add a comment here about what situations would debianVersion not exist, and why it's not an error.

File: vulnfeeds/cmd/debian/main.go [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+				if debianVersion == "" {
+					continue
+				}
+
+				pkgInfo := vulns.PackageInfo{
+					PkgName:   pkgName,
+					Ecosystem: "Debian" + debianVersion,
```
Comment (another-rex) [inline vulnfeeds/cmd/debian/main.go:73]: Does this need a `:` between Debian and debianVersion?

File: vulnfeeds/cmd/debian/main.go [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+						continue
+					}
+					pkgInfo.VersionInfo = cves.VersionInfo{
+						AffectedVersions: []cves.AffectedVersion{{Fixed: release.FixedVersion}},
+					}
+				}
+				pkgInfo.EcosystemSpecific = make(map[string]string)
+				pkgInfo.EcosystemSpecific["urgency"] = release.Urgency
+				pkgInfos = append(pkgInfos, pkgInfo)
+				osvPkgInfos[cveId] = pkgInfos
```
Comment (another-rex) [inline vulnfeeds/cmd/debian/main.go:86]: nit: this line should be outside this for loop (1 line down)

File: vulnfeeds/cmd/debian/main.go [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+				}
+				if release.Status == "resolved" {
+					if release.FixedVersion == "0" { // not affected
+						continue
+					}
+					pkgInfo.VersionInfo = cves.VersionInfo{
+						AffectedVersions: []cves.AffectedVersion{{Fixed: release.FixedVersion}},
+					}
+				}
+				pkgInfo.EcosystemSpecific = make(map[string]string)
```
Comment (another-rex) [inline vulnfeeds/cmd/debian/main.go:83]: nit: This can be moved into the initial creation of the struct on line 74 ish

**Review by another-rex — COMMENTED** [review]
Looks good!, added some comments.

File: vulnfeeds/cmd/debian/main.go (hunk 2 above)
Comment (hogo6002) [reply vulnfeeds/cmd/debian/main.go:67]: Thanks Rex! Added a comment to the getDebianVersion function.

File: deployment/build-and-stage.yaml
Comment (michaelkedar) [inline deployment/build-and-stage.yaml:1]: You'll need to add `debian-cve-convert=gcr.io/oss-vdb/debian-cve-convert:$COMMIT_SHA` to the [cloud deploy step](https://github.com/google/osv.dev/blob/86bc8bd39e1e45ecedfededdece9df134b3bc29b/deployment/build-and-stage.yaml#L258-L269)

**Review by michaelkedar — COMMENTED** [review]
Infra stuff looks good, just one thing:

**Review by another-rex — APPROVED** [review]

File: vulnfeeds/cmd/debian/main.go [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+			seriesIndex = i
+		}
+	}
+
+	if seriesIndex == -1 || versionIndex == -1 {
+		Logger.Fatalf("Failed to get Debian release info: %s", err)
+	}
+
+	for _, row := range data[1:] {
+		if row[seriesIndex] == "experimental" || row[seriesIndex] == "sid" {
```
Comment (hogo6002) [inline vulnfeeds/cmd/debian/main.go:70]: @another-rex do we want to keep sid records? I noticed that we kept them and named as "unstable" for [convert_debian](https://github.com/google/osv.dev/blob/2ed3d46bd5c433bbbb06b72d7c0c171546646550/vulnfeeds/tools/debian/debian_converter/convert_debian.py#L173) one, but I couldn't find any [debian:unstable](https://osv.dev/list?ecosystem=Debian&q=Debian%3Aunstable) packages on osv.dev

File: deployment/clouddeploy/gke-workers/environments/oss-vdb/debian-cve-convert.yaml [hunk 7]
```diff
@@ -0,0 +1,16 @@
+apiVersion: batch/v1
```
Comment (oliverchang) [inline deployment/clouddeploy/gke-workers/environments/oss-vdb/debian-cve-convert.yaml:1]: Should we hold off on prod for now, and do test to validate the results first?

File: vulnfeeds/cmd/debian/main.go [hunk 8]
```diff
@@ -0,0 +1,158 @@
...
+	osvPkgInfos := make(map[string][]vulns.PackageInfo)
+	for pkgName, pkg := range debianData {
+		for cveId, cve := range pkg {
+			var pkgInfos []vulns.PackageInfo
+			if value, ok := osvPkgInfos[cveId]; ok {
+				pkgInfos = value
+			}
+
+			for releaseName, release := range cve.Releases {
+				if release.Urgency == "not yet assigned" || release.Urgency == "end-of-life" {
```
Comment (oliverchang) [inline vulnfeeds/cmd/debian/main.go:92]: Can you add some comments here to explain the cases? Ideally also some documentation as well if you can find any.

File: vulnfeeds/cmd/debian/main.go (hunk 6 above)
Comment (hogo6002) [reply vulnfeeds/cmd/debian/main.go:71]: Discussed with Rex offline, we will ignore unstable ones first

File: vulnfeeds/cmd/debian/main.go [hunk 9]
```diff
@@ -0,0 +1,164 @@
...
+
+	return osvPkgInfos
+}
+
+func writeToOutput(cvePkgInfos map[string][]vulns.PackageInfo) {
+	for cveId := range cvePkgInfos {
+		pkgInfos := cvePkgInfos[cveId]
+		file, err := os.OpenFile(path.Join(debianOutputPathDefault, cveId+".debian.json"), os.O_CREATE|os.O_RDWR, 0644)
+		if err != nil {
+			Logger.Fatalf("Failed to create/write osv output file: %s", err)
```
Comment (oliverchang) [inline vulnfeeds/cmd/debian/main.go:138]: nit: better to return an error instead, and let the caller handle the error.

File: vulnfeeds/cmd/debian/main.go [hunk 10]
```diff
@@ -0,0 +1,164 @@
...
+		}
+		_ = file.Close()
+	}
+}
+
+// downloadDebianSecurityTracker download Debian json file
+func downloadDebianSecurityTracker() DebianSecurityTrackerData {
+	res, err := http.Get(debianSecurityTrackerURL)
+	if err != nil {
+		Logger.Fatalf("Failed to get Debian Security Tracker json: %s", err)
```
Comment (oliverchang) [inline vulnfeeds/cmd/debian/main.go:154]: same comment here re error handling.

**Review by oliverchang — APPROVED** [review]
LGTM with some minor nits.

**Review by oliverchang — APPROVED** [review]

**Pull request merged**